roles/splunk/tasks/configure_auditd.yml

```diff
@@ -0,0 +1,45 @@
+---
+- name: "get {{ splunk_nix_user }} user uid"
+  getent:
+    database: passwd
+    key: "{{ splunk_nix_user }}"
+
+# - debug:
+#     msg: "{{ splunk_nix_user }} user id = {{ ansible_facts.getent_passwd[splunk_nix_user][1] }}"
+
+- name: auditd - set 20-splunk.rules
+  become: true
+  template:
+    src: 20-splunk.rules.j2
+    dest: /etc/audit/rules.d/20-splunk.rules
+    mode: 0600
+    owner: root
+    group: root
+  register: splunk_rule
+
+- name: Get auditd enabled level  # noqa 301 306
+  become: true
+  shell: auditctl -s | grep enabled | cut -d" " -f2
+  changed_when: false
+  check_mode: false
+  register: auditctl_enabled
+
+# restart auditd if not immutable
+- name: restart auditd if not immutable  # noqa no-handler
```

Comment:
when `splunk_rule is changed` `auditd` should be able to be reloaded without a restart, only enabling auditd needs a restart.
The restart and reload should probably be two separate tasks.

Reply:
reloading auditd is only possible if it's not configured to be immutable. It can still be reloaded, but no changes are applied until reboot. Because of this we don't want ansible to report "changed" on a task when it never successfully applied anything giving the user a false sense of completion.

We also can't in good conscious be applying reboots to machines. This is the best approach we've found which is to apply if we can (auditd is not immutable) or notify user via debug msg that a reboot is required (auditd is immutable). 

We've had these tasks in place for a long time now and they're really solid.